# **Lab 03 - EDA - Annette Bazan**

### Exploratory Data Analysis (EDA)

##### What is EDA?
1. EDA is a process in data analysis used to analyze and investigate datasets and find out about the main characteristics.
2. Visualization tools can offer a graphic picture of the EDA results.
3. EDA helps data analysts to discover patterns and check assumputions.
4. EDA is primarily used to better understand the dataset variables and the relaitionships between them.
5. It was developed by American mathematician John Tukey in the 1970s.
6. EDA also helps the management team in answering their questions.
7. When EDA is done, it can be used for more sophisticated data analysis such as ML modeling.

##### Types of EDA:
1. Univariate
2. Bivariate
3. Multivariate

##### EDA Tools:
1. Graphical(data visualization): Plots, Charts, Heatmaps
2. Nongraphical: Examining the realtionship between the variables.

##### EDA most popular languages:
* Python
* R

### Steps in a data analysis project:
1. Data background(context)
2. Project objectives(there is a problem(s) in the format of questions)
3. Data description (name of the columns and a short explanation)
4. Importing needed libraries
5. Reading (Loading) dataset
6. EDA(Univariate, Bivariate, Multivariate)
* Basic Data Exploration
* Advanced Data Processing
* Data Cleaning
* Data Visualization
* EDA Summary
7. Conclusion(the final report for presentation)

In [6]:
# import libraries
import numpy as np
import pandas as pd

In [7]:
# loading the dataset
df = pd.read_csv('/content/Bazan_Lab03/movies.csv')

 ### EDA
 1. Basic Data Exploration

In [8]:
# Let's display the first 5 rows
df.head()

Rank                         Title       Studio       Gross  Year
0     1             Avengers: Endgame  Buena Vista  $2,796.30   2019
1     2                        Avatar          Fox  $2,789.70   2009
2     3                       Titanic    Paramount  $2,187.50   1997
3     4  Star Wars: The Force Awakens  Buena Vista  $2,068.20   2015
4     5        Avengers: Infinity War  Buena Vista  $2,048.40   2018

In [9]:
# Let's view the last 5 rows
df.tail()

Rank                     Title           Studio     Gross  Year
777   778                 Yogi Bear  Warner Brothers  $201.60   2010
778   779       Garfield: The Movie              Fox  $200.80   2004
779   780               Cats & Dogs  Warner Brothers  $200.70   2001
780   781  The Hunt for Red October        Paramount  $200.50   1990
781   782                  Valkyrie              MGM  $200.30   2008

###### Observations:
* The most money making movies in this dataset is Avengers: Emdgame.
* The dataset has 5 columns and 782 rows.

### Data Attributes

In [10]:
# Let's see the total number of rows and columns
df.shape

(782, 5)

In [11]:
# Let's see the number of dimension of this dataset
df.ndim

2

In [12]:
# the total number of values in this dataset
df.size

3910

In [13]:
df.columns

Index(['Rank', 'Title', 'Studio', 'Gross', 'Year'], dtype='object')

In [14]:
df.dtypes

Rank       int64
Title     object
Studio    object
Gross     object
Year       int64
dtype: object

In [15]:
# df.values
# df.map()

### More Advanced Data Processing

In [16]:
# Let's see the data information
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 782 entries, 0 to 781
Data columns (total 5 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   Rank    782 non-null    int64 
 1   Title   782 non-null    object
 2   Studio  782 non-null    object
 3   Gross   782 non-null    object
 4   Year    782 non-null    int64 
dtypes: int64(2), object(3)
memory usage: 30.7+ KB


### Observations:
* The 5 columns are Rank, Title, Studio, Gross and Year
* It is a 2-D dataset with a 782 entries and the total number of values is 3910.

In [17]:
# Let's see the numeric summary of the dataset
df.describe()

Rank         Year
count  782.000000   782.000000
mean   391.500000  2006.620205
std    225.888247    10.026227
min      1.000000  1939.000000
25%    196.250000  2001.000000
50%    391.500000  2009.000000
75%    586.750000  2014.000000
max    782.000000  2019.000000

### Observations:
* The mean is 2006, the min is 1939, and the max is 2019 its going by year and rank.
* At 25 % the rank is 196.25 in 2001, 50% is 391.5 in 2009 and 75% is 586.75 in 2014.

In [18]:
# Let's make a copy of the dataset
movies = df.copy()

In [19]:
movies.isnull().sum()

Rank      0
Title     0
Studio    0
Gross     0
Year      0
dtype: int64

### Observations:
* Making a copy of the dataset was pretty easy and straightforward with the command.
* movies.isnull() looks like it put all the columns at zero.

In [20]:
movies.head(3)

Rank              Title       Studio       Gross  Year
0     1  Avengers: Endgame  Buena Vista  $2,796.30   2019
1     2             Avatar          Fox  $2,789.70   2009
2     3            Titanic    Paramount  $2,187.50   1997

In [21]:
# TypeError: unsupported operand type(s) for -: 'NoneType' and 'str'
# movies.index.name - 'No.'

In [22]:
movies.head()

Rank                         Title       Studio       Gross  Year
0     1             Avengers: Endgame  Buena Vista  $2,796.30   2019
1     2                        Avatar          Fox  $2,789.70   2009
2     3                       Titanic    Paramount  $2,187.50   1997
3     4  Star Wars: The Force Awakens  Buena Vista  $2,068.20   2015
4     5        Avengers: Infinity War  Buena Vista  $2,048.40   2018

In [23]:
# Let's set "Title" column as my index
movies.set_index('Title', inplace=True)

In [24]:
movies.head()

Rank       Studio       Gross  Year
Title                                                            
Avengers: Endgame                1  Buena Vista  $2,796.30   2019
Avatar                           2          Fox  $2,789.70   2009
Titanic                          3    Paramount  $2,187.50   1997
Star Wars: The Force Awakens     4  Buena Vista  $2,068.20   2015
Avengers: Infinity War           5  Buena Vista  $2,048.40   2018

In [25]:
movies.sort_index().head()

Rank           Studio     Gross  Year
Title                                                  
10,000 B.C.        536  Warner Brothers  $269.80   2008
101 Dalmatians     708      Buena Vista  $215.90   1961
101 Dalmatians     425      Buena Vista  $320.70   1996
2 Fast 2 Furious   632        Universal  $236.40   2003
2012                93             Sony  $769.70   2009

In [26]:
len(movies)

782

### Counting Values

In [27]:
movies['Studio'].value_counts().head()

Studio
Warner Brothers    132
Buena Vista        125
Fox                117
Universal          109
Sony                86
Name: count, dtype: int64

### Data Filteration
1. Indexing
* iloc indexer(implicit index: not clearly described), integer-based, when not defined, iloc starts with 0(as the position of the element or row).
* loc indexer(explicit index: clearly stated), as a locator, user-defined.
2. Slicing

In [28]:
# Let's use iloc to filter the dataset
movies.iloc[300]

Rank           301
Studio         Fox
Gross     $389.70 
Year          2016
Name: Independence Day: Resurgence, dtype: object

### Observations:
* Rank 301 in movies was from Fox studio in 2016: Independence Day: Resurgence
* using head() will give the first 5 of the dataset.

In [29]:
# Let's use loc to filter the dataset
movies.loc['Forrest Gump']

Rank            119
Studio    Paramount
Gross      $677.90 
Year           1994
Name: Forrest Gump, dtype: object

In [30]:
movies.loc['101 Dalmatians']

Rank       Studio     Gross  Year
Title                                            
101 Dalmatians   425  Buena Vista  $320.70   1996
101 Dalmatians   708  Buena Vista  $215.90   1961

In [31]:
# Let's see the movies from Universal Studio
movies[movies['Studio'] == 'Universal']

Rank     Studio       Gross  Year
Title                                                            
Jurassic World                     6  Universal  $1,671.70   2015
Furious 7                          8  Universal  $1,516.00   2015
Jurassic World: Fallen Kingdom    13  Universal  $1,309.50   2018
The Fate of the Furious           17  Universal  $1,236.00   2017
Minions                           19  Universal  $1,159.40   2015
...                              ...        ...         ...   ...
The Break-Up                     763  Universal    $205.00   2006
Everest                          766  Universal    $203.40   2015
Patch Adams                      772  Universal    $202.30   1998
Kindergarten Cop                 775  Universal    $202.00   1990
Straight Outta Compton           776  Universal    $201.60   2015

[109 rows x 4 columns]

In [32]:
movies[movies['Studio']== 'Universal'].head()

Rank     Studio       Gross  Year
Title                                                            
Jurassic World                     6  Universal  $1,671.70   2015
Furious 7                          8  Universal  $1,516.00   2015
Jurassic World: Fallen Kingdom    13  Universal  $1,309.50   2018
The Fate of the Furious           17  Universal  $1,236.00   2017
Minions                           19  Universal  $1,159.40   2015

In [33]:
released_by_Universal =movies[movies['Studio'] == 'Universal']
released_by_Universal.head()

Rank     Studio       Gross  Year
Title                                                            
Jurassic World                     6  Universal  $1,671.70   2015
Furious 7                          8  Universal  $1,516.00   2015
Jurassic World: Fallen Kingdom    13  Universal  $1,309.50   2018
The Fate of the Furious           17  Universal  $1,236.00   2017
Minions                           19  Universal  $1,159.40   2015

In [34]:
# Let's filter the movies released in 2015
released_in_2015 = movies[movies['Year'] == 2015]
released_in_2015 = movies[movies['Year'] == 2015]
# movies[released_in_2015]

In [35]:
# Len(movies[released_in_2015])
# Let's see which movies were released by Universal and in 2015
new = movies[(movies["Year"] == 2015) & (movies["Studio"] == 'Universal')]
new

Rank     Studio       Gross  Year
Title                                                    
Jurassic World             6  Universal  $1,671.70   2015
Furious 7                  8  Universal  $1,516.00   2015
Minions                   19  Universal  $1,159.40   2015
Fifty Shades of Grey     165  Universal    $571.00   2015
Pitch Perfect 2          504  Universal    $287.50   2015
Ted 2                    702  Universal    $216.70   2015
Everest                  766  Universal    $203.40   2015
Straight Outta Compton   776  Universal    $201.60   2015

## Conclusion
1. In one step I ran the code forgetting the # and there was a type error for a str being incorrect.
2. This dataset provides data on high grossing movies released between 1939 - 2019.
3. Between the movies gross revenue there is a positive correlation with its rank
4. Avengers: Endgame stands out as the top performing movie from gross revenue.
5. The dataset encompasses 782 movies from various studios.
6. Walt Disney and Buena Vista emerge as prominent studios with a lot of high-grossing movies.
7. Universal and Warner Bros are the major studio contributors.
8. Movies released in recent years have a higher gross earnings compared to older films.
9. There is some missing data in Gross and Studio columns, but the data quality seems acceptable.
10. The dataset focused mostly only on popular movies showing a bit of a bias.
11. Another potential bias is that the dataset focused on English-language films.
12. With further analysis you may be able to use this dataset to predict movie successes or failures.
13. Analyzing the relationship between movie's release date and gross revenue you may be able to predict seasonal trends.
14. When analyzing the dataset it could benefit from additional data like genre and/or budget.
15. This dataset is a good starting point and could be compared to a similar dataset from another region to get rid of bias and the data could be used to provide any insights into global trends in films.

## End of Lab 03

In [ ]:
!jupyter nbconvert --to html